src/pugixml.cpp

```diff
@@ -3160,6 +3177,8 @@ PUGI__NS_BEGIN
 
 			if (declaration ? PUGI__OPTSET(parse_declaration) : PUGI__OPTSET(parse_pi))
 			{
+				char_t ch = 0;
```

Comment:
ch variable is special in that it's used by ENDSEG macro. Right now the closest invocation of this macro is inside this scope but the intent is to have ch variable in every function that uses the macro. So this change makes future modifications to this function slightly harder - please put this variable back.